addons/io_hubs_addon/components/components_registry.py

```diff
@@ -34,6 +34,35 @@ def get_components_in_dir(dir):
     return sorted(components)
 
 
+def get_user_component_names():
+    component_names = []
+    from ..preferences import get_addon_pref
+    addon_prefs = get_addon_pref(bpy.context)
+    for _, entry in enumerate(addon_prefs.user_components_paths):
+        if entry.path:
```

Comment:
I think a check should be added here to make sure the path exists, otherwise the whole add-on can be broken.
```suggestion
        if entry.path and os.path.isdir(entry.path):
```

Reply:
Added